scripts/all-helpers.sh

```diff
@@ -254,17 +254,30 @@ helper_armc6_build_test()
     FLAGS="$1"
 
     msg "build: ARM Compiler 6 ($FLAGS)"
-    make clean
-    ARM_TOOL_VARIANT="ult" CC="$ARMC6_CC" AR="$ARMC6_AR" CFLAGS="$FLAGS" \
-                    WARNING_CFLAGS='-Werror -xc -std=c99' make lib
-
-    msg "size: ARM Compiler 6 ($FLAGS)"
-    "$ARMC6_FROMELF" -z library/*.o
-    if [ -n "${PSA_CORE_PATH}" ]; then
-        "$ARMC6_FROMELF" -z ${PSA_CORE_PATH}/*.o
-    fi
-    if [ -n "${BUILTIN_SRC_PATH}" ]; then
-        "$ARMC6_FROMELF" -z ${BUILTIN_SRC_PATH}/*.o
+
+    if in_mbedtls_repo; then
+        make clean
+        ARM_TOOL_VARIANT="ult" CC="$ARMC6_CC" AR="$ARMC6_AR" CFLAGS="$FLAGS" \
+                         WARNING_CFLAGS='-Werror -xc -std=c99' make lib
+        msg "size: ARM Compiler 6 ($FLAGS)"
+        "$ARMC6_FROMELF" -z library/*.o
+        if [ -n "${PSA_CORE_PATH}" ]; then
+            "$ARMC6_FROMELF" -z ${PSA_CORE_PATH}/*.o
+        fi
+        if [ -n "${BUILTIN_SRC_PATH}" ]; then
+            "$ARMC6_FROMELF" -z ${BUILTIN_SRC_PATH}/*.o
+        fi
+    else
+        cd $TF_PSA_CRYPTO_ROOT_DIR
+        rm -rf $OUT_OF_SOURCE_DIR
+        mkdir $OUT_OF_SOURCE_DIR
+        cd $OUT_OF_SOURCE_DIR
+        cmake -DCMAKE_SYSTEM_NAME="Generic" -DCMAKE_SYSTEM_PROCESSOR="cortex-m0" \
```

Comment:
For some reason, when using CMake the generic system processor `armv7-m` is not an option even though the ARM compiler 6 user guide specifies it as. This may just be a CMake toolchain issue where a default file is being used to check compiler options but I was previously testing this inside it's own toolchain file and experiencing the same error.

Therefore my proposed workaround is to set the system processor to the `cortex-m0`, this should be fine since the `cortex-m0` implementation(when compared to others) is minimal in feature set, so building for subsequent revisions should work.